### DISCUSSION TITLE
Перенос миниатюр между компьютерами

### DISCUSSION THREAD

**Discussion by udavst** [k-ivan/visual-bookmarks-chrome#193] [Q&A]
Как перенести миниатюры между компьютерами? Для отображения выбрана одна и та же папка закладок, многие иконки присвоены вручную, на другом компьютере они не появляются, может какую-то папку из профиля перенести?

**Comment by k-ivan** [comment]
Такой возможности, к сожалению, нет. Если только пытаться пробовать переносить папку с данными браузера на другое устройство, но не пробовал, поэтому никаких гарантий, что это сработает.

**Reply by kobav14** [reply]
Посмотрел во внутренних пакпках chrome и не обнаружил ничего, напоминающее файлы изображений, скорее всего это закрыли в обновленных версиях chrome
Можно ли вывести в контекстном меню Thumbnails наряду с "Upload from computer" пункт "Download to computer" дав возможность скачать, и/или "открыть миниатюру в новой вкладке" ссылку вида blob:chrome-extension://hnhinkooemphbnnpmjcnjgmafjhlofci/5d45be51-d339-44ae-8ade-3b4095ea8e91
Это минимально что хочется иметь, если не замахиваться на полноценный импорт/экспорт миниатюр

**Reply by k-ivan** [reply]
Посмотрите чуть ниже один из моих [комментариев](https://github.com/k-ivan/visual-bookmarks-chrome/discussions/193#discussioncomment-11481958) (на примере chrome браузера). Доступ к ним открыт, так как это пользовательские файлы. Эти файлы не будут иметь расширения в файловой системе, то есть, они не будут помечены как изображения, но это изображения и их можно открыть в любом приложении просмотра изображений.

Я обдумаю Ваше предложение, по поводу добавления пункта скачивания в контекстное меню.

**Reply by kobav14** [reply]
вот примеры файлов c:\Users\{USER}\AppData\Local\Google\Chrome\User Data\Default\IndexedDB\chrome-extension_hnhinkooemphbnnpmjcnjgmafjhlofci_0.indexeddb.blob\1\
добавил расширения png чтобы можно было прикрепить сюда

**Reply by k-ivan** [reply]
Скорее всего Вы что-то упускайте из виду, изображения хранятся в расширении `.webp`. Но сейчас попробовал скачать его как `.png` и никаких проблем, так же открывается успешно.

Вы попробуйте прямо в барузере ввести ссылку выше, и там открыть один из доступных файлов, это будет работать и в проводнике и в браузере.

**Reply by k-ivan** [reply]
Если все правильно, то увидите в браузере вот такую картину
![image](https://github.com/user-attachments/assets/ef23a060-734a-42cc-8f51-54be3010fdbb)
Если перейдете по одной из ссылок, в браузере откроется изображение

**Reply by kobav14** [reply]
да, всё верно, спасибо. не то расширение смотрел..

**Comment by udavst** [comment]
Всю папку с профилем? Или достаточно какой-то одной папки (например с хранилищем расширения, если такое есть)? Весь профиль уж очень не хочется, а если есть, хоть призрачный шанс, что поможет какая-то папка оттуда, было бы здорово.

**Comment by k-ivan** [comment]
Сами изображения хранятся в папке (в примере профиля браузера по умолчанию на Windows 11). Внутри подпапок будут полноценные изображения
```sh
C:\Users\{имя_пользователя}\AppData\Local\Google\Chrome\User Data\Default\IndexedDB\chrome-extension_jdbgjlehkajddoapdgpdjmlpdalfnenf_0.indexeddb.blob\
```
и так же есть папка c сервисной информацией и базой данных indexDB
```
C:\Users\{имя_пользователя}\AppData\Local\Google\Chrome\User Data\Default\IndexedDB\chrome-extension_jdbgjlehkajddoapdgpdjmlpdalfnenf_0.indexeddb.leveldb
```

Но надо пробовать, гарантии дать не могу, и за работоспособность ничего обещать не могу.

**Comment by k-ivan** [comment]
@udavst, но перенос самих изображений и базы не поможет. Основная проблема сейчас в том, что изображения к закладкам я привязываю через id закладок, но на другом устройстве одни и теже закладки будут иметь разные идентификаторы. А как обойти данную проблему, пока идей нет.

**Comment by udavst** [comment]
Понял, тогда достаточно знать папку с картинками, потом на другом компе их заново прилепить.
Только вот ID  (папка) расширения в Crome не совпадает видимо с Firefox, подстроки jdbgjlehkajddoapdgpdjmlpdalfnenf найти не удалось.

**Comment by k-ivan** [comment]
Да, все верно, в фаерфоксе будет другой id. Попробуйте открыть страницу настроек, в адресной строке у Вас отобразится ID расширения.
![image](https://github.com/user-attachments/assets/c725c86f-8d1e-4584-8d04-985a90a4b8d7)

**Comment by k-ivan** [comment]
Вот путь до файлов изображений
```
C:/Users/{имя_пользователя}/AppData/Roaming/Mozilla/Firefox/Profiles/jrgn4g6q.default-release/storage/default/moz-extension+++7f5e2951-cc38-45bc-ab08-e2d67f759240/idb
```

Файлики будут без расширения, но это картинки, пусть Вас это не смущает

**Reply by k-ivan** [reply]
Ну и еще один минус это имена файлов, то есть в проводнике не понятно, какая картинка от какого сайта.

**Comment by udavst** [comment]
Да-да, я разобрался, спасибо, всё в порядке.
Выведу их все миниатюрами, и заберу нужные )

**Reply by k-ivan** [reply]
Отлично!
